R/fdbk_asdataframe.R

```diff
@@ -1203,6 +1261,133 @@ fdbk_dt_binning <- function(DT,varToBin="level",binLower,binUpper){
+#' obs = T1
+#' veri_data = T2
+#' veri_forecast_time = 24
+#' veri_initial_date = 2015110900
+#' time = -720
+#' lat = 46.812
+#' lon = 6.943
+#' varno = 2
+#' veri_model = "COSMO"
+#' plevel = p
+#' ident = 6610
+#' levels = c(1000, 975, 950, 925, 900, 875, 850, 800, 750, 700, 650, 600, 550, 500, 450, 400, 350, 300, 250) # Levels on which to interpolate
+#' DT = data.frame(obs,veri_data,veri_forecast_time,veri_initial_date,time,lat,lon,varno,veri_model,plevel,ident)
+#' DT           = fdbk_dt_interpolate(DT,varToInter=c("obs","veri_data"), levelToInter = "plevel", interLevels = levels) # interpolate DT
+#'
+#' data1 = melt(data.frame(T1,p),id="T1") # Data for the standard atmosphere temperature profile
+#' data2 = melt(data.frame(T2=DT$obs,DT$plevel),id="T2") # Interpolation of data1 
+#' 
+#' 
+#' plot = ggplot() + geom_point(data=data1,aes(x=T1,y=value,colour=variable)) + geom_point(data=data2,aes(x=T2,y=value,colour=variable))+scale_y_reverse()+
+#'   xlab("T [K]") + ylab("pressure [hPa]")+  scale_colour_manual(name="Temperature",values=c("red","blue"),labels=c("Interpolation","Standard atmosphere"))+theme(legend.position = "top")
+#' print(plot) # plot of the Standard atmosphere profile and its interpolation 
+#' 
+#' allscores = fdbk_dt_verif_continuous(DT,strat=c("varno","veri_model","plevel") ) # Data table with scores
+#' 
+#' data3 = melt(data.frame(Bias,p),id="Bias") # Bias calculated directly from the standard atmosphere and model output profiles
+#' ME = allscores[allscores$scorename=="ME"]$scores # scores calculated with fdbk_dt_verif_continuous on interpolation levels
+#' ME_levels = allscores[allscores$scorename=="ME"]$plevel # interpolation levels
+#' data4 = melt(data.frame(ME,ME_levels),id="ME")
+#' plot2 = ggplot() + geom_point(data=data3,aes(x=Bias,y=value,colour=variable)) + geom_point(data=data4,aes(x=ME,y=value,colour=variable))+scale_y_reverse()+
+#'   xlab("T bias [K]") + ylab("pressure [hPa]")+scale_colour_manual(name="Bias",values=c("red","blue"),labels=c("Interpolation","Standard atmosphere"))+theme(legend.position = "top")
+#' print(plot2) # plot of the bias calculated directly from the profiles and the bias from the interpolation
+
+fdbk_dt_interpolate <- function(DT,varToInter=c("obs","veri_data"), levelToInter = "plevel", interLevels = levels, varno="varno"){
+  
+  #OneSounding = data.frame(,nrow=1,ncol=dim(DT)[2])
+  ncol = dim(DT)[2]
+  n = dim(DT)[1]
+  OneSounding = as.data.frame(matrix(,nrow=1,ncol)) # vector to store one single sounding at a time and interpolate it
+  Colnames = colnames(DT)
+  colnames(OneSounding) = Colnames
+  OneSounding[1,] = DT[1,] # Store the first row of DT in the first row of OneSounding
+  newDT = c()
+  k = 1
+  for (i in 2:n){
+    
+    soundingDone = TRUE # Boolean if i correspond to the end of a souding 
+    if (DT$ident[i]==DT$ident[i-1] & DT$veri_forecast_time[i]==DT$veri_forecast_time[i-1] & DT$veri_initial_date[i]==DT$veri_initial_date[i-1] & DT$time[i]==DT$time[i-1]){
```

Comment:
Could this be rewritten like so?
checknames <- c("ident", "veri_forecast_time", "veri_initial_date", ...)
if (all(DT[i,checknames] == DT[i-1,checknames])){
...